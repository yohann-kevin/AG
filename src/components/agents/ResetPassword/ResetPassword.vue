<template>
  <div class="reset-password">
    <h2>Réinitialisation du mot de passe</h2>
    <div class="reset-password-container-alert">
      <v-alert
        dense
        text
        dismissible
        elevation="15"
        type="success"
        v-model="successAlert"
        class="reset-password-alert"
      >
        Le mail de réinitialisation à bien été envoyer ! Si vous ne trouver pas le mail en question vérifier dans vos spam.
      </v-alert>
    </div>
    <v-form
      v-model="valid"
      class="reset-password-form"
      ref="reset-password-form"
    >
      <v-text-field
        v-model="email"
        color="black"
        :rules="emailRules"
        type="email"
        label="Courriel"
        class="reset-password-input"
        required
      />

      <div class="reset-password-btn">
        <v-btn
          text
          :disabled="!valid"
          @click="resetPassword"
        >
          Envoyer
        </v-btn>
        <v-btn
          text
          @click="resetForm"
        >
          Annuler
        </v-btn>
      </div>
    </v-form>
  </div>
</template>

<script>
export default {
  data: () => ({
    successAlert: false,
    valid: false,
    email: '',
    emailRules: [ v => /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/.test(v) || 'Adresse email invalide' ],
  }),
  methods: {
    resetForm() {
      this.$refs['reset-password-form'].reset();
    },
    resetPassword() {
      const data = {
        agent_email: this.email
      };

      const config = {
        method: 'post',
        // eslint-disable-next-line no-undef
        url: process.env.VITE_APP_API_URL + 'renew_passwords',
        headers: { 
          'Content-Type': 'application/json'
        },
        data : data
      };

      this.$axios(config).then(response => {
        console.log(response.data);
      }).catch(error => {
        console.log(error);
      });

      this.successAlert = true;
    }
  },
}
</script>

<style scoped>
.reset-password {
  width: 100%;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.reset-password h2 {
  width: 100%;
  text-align: center;
  margin: 10px;
}

.reset-password-container-alert {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.reset-password-alert {
  width: 40%;
}

.reset-password-form {
  width: 40%;
}

.reset-password-input {
  margin: 10px;
}

.reset-password-btn {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
</style>
