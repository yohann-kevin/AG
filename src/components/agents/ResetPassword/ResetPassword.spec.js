import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import ResetPassword from './ResetPassword.vue';

describe('ResetPassword', () => {
  const initWrapper = () => {
    const localVue = createLocalVue();
    const vuetify = new Vuetify();
    return shallowMount(ResetPassword, {
      localVue,
      vuetify
    });
  };

  it('should test component render correctly', () => {
    const wrapper = initWrapper();

    const title = wrapper.find('h2');
    expect(title.text()).toContain('RÃ©initialisation du mot de passe');
    expect(wrapper).toMatchSnapshot();
  });
});
