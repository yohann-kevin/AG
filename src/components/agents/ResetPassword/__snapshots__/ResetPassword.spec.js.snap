// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should test component render correctly 1`] = `
<div class="reset-password">
  <h2>Réinitialisation du mot de passe</h2>
  <div class="reset-password-container-alert">
    <v-alert dense="" text="" dismissible="" elevation="15" type="success" class="reset-password-alert">
      Le mail de réinitialisation à bien été envoyer ! Si vous ne trouver pas le mail en question vérifier dans vos spam.
    </v-alert>
  </div>
  <v-form class="reset-password-form">
    <v-text-field color="black" rules="v => /^[\\w-\\\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$/.test(v) || 'Adresse email invalide'" type="email" label="Courriel" required="" class="reset-password-input"></v-text-field>
    <div class="reset-password-btn">
      <v-btn text="" disabled="true">
        Envoyer
      </v-btn>
      <v-btn text="">
        Annuler
      </v-btn>
    </div>
  </v-form>
</div>
`;
